### PR TITLE
Add machine readable output (-o flag) to kn source ping describe

### LINKED ISSUE #1098: Add machine readable output (-o flag) to kn source ping describe
use case: `kn source binding describe NAME -oyaml`

see `kn service describe NAME -oyaml` for ref

/kind good-first-issue

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

## Changes

*Add machine readable output (-o flag) to kn source ping describe

## Reference

Fixes #1098

### CODE REVIEW

File: pkg/kn/commands/source/ping/describe.go [hunk 1]
```diff
@@ -26,15 +30,23 @@ import (
 	"knative.dev/client/pkg/printers"
 )
 
+var describeExample = `
+  # Describe a ping source with name 'myping'
+  kn source ping describe myping
+
+  # Describe a ping source with name 'myping' in YAML format
+  kn source ping describe myping -o yaml`
```
Comment (navidshaikh) [inline pkg/kn/commands/source/ping/describe.go:38]:
```suggestion
  # Describe a ping source 'myping'
  kn source ping describe myping

  # Describe a ping source 'myping' in YAML format
  kn source ping describe myping -o yaml`
```

File: pkg/sources/v1alpha2/ping_client.go [hunk 2]
```diff
@@ -87,7 +87,15 @@ func (c *pingSourcesClient) DeletePingSource(name string) error {
 }
 
 func (c *pingSourcesClient) GetPingSource(name string) (*v1alpha2.PingSource, error) {
-	return c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	source, err := c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
```
Comment (navidshaikh) [inline pkg/sources/v1alpha2/ping_client.go:92]:
```suggestion
		return nil, knerrors.GetError(err)
```

(import it as `	knerrors "knative.dev/client/pkg/errors"`)

Comment (navidshaikh) [reply pkg/sources/v1alpha2/ping_client.go:92]: we'd need to align catching this error at individual resource-client level across the code, lets begin from here, I will open another issue with details

**Review by navidshaikh — APPROVED** [review]
a few nits, but rest lgtm

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -17,6 +17,10 @@
 |===
 | | Description | PR
 
+| 🐛
```
Comment (navidshaikh) [inline CHANGELOG.adoc:20]:
```suggestion
| 🎁
```

File: CHANGELOG.adoc [hunk 4]
```diff
@@ -17,6 +17,10 @@
...
+| Add machine readable output (-o flag) to kn source ping describe
+| https://github.com/knative/client/pull/1150[#1150]
+
 | 🐛
```
Comment (navidshaikh) [inline CHANGELOG.adoc:24]:
```suggestion
| 🎁
```

Comment (navidshaikh) [reply CHANGELOG.adoc:24]: here as well :-)

**Review by navidshaikh — COMMENTED** [review]
🎁

**Comment by arghya88** [comment]
@navidshaikh changes done.PTAL

File: pkg/kn/commands/source/ping/describe.go [hunk 5]
```diff
@@ -16,7 +16,11 @@ package ping
...
 import (
 	"errors"
+	"fmt"
 	"sort"
+	"strings"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 
 	"github.com/spf13/cobra"
 	v1alpha2 "knative.dev/eventing/pkg/apis/sources/v1alpha2"
```
Comment (navidshaikh) [inline pkg/kn/commands/source/ping/describe.go:26]: can group these imports together

File: pkg/sources/v1alpha2/ping_client.go [hunk 6]
```diff
@@ -87,7 +89,15 @@ func (c *pingSourcesClient) DeletePingSource(name string) error {
...
 func (c *pingSourcesClient) GetPingSource(name string) (*v1alpha2.PingSource, error) {
-	return c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	source, err := c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, knerrors.GetError(err)
+	}
+	err = updateSourceGVK(source)
+	if err != nil {
+		return nil, knerrors.GetError(err)
+	}
```
Comment (navidshaikh) [inline pkg/sources/v1alpha2/ping_client.go:99]:
```suggestion
	err = updateSourceGVK(source)
	if err != nil {
		return nil, err
	}
```

Comment (navidshaikh) [reply pkg/sources/v1alpha2/ping_client.go:99]: this is only client-side operation and we can simply return the `err` here

Although `knerrors.GetError` would just get back with original error, the idea behind passing `err` through `knerrors.GetError` is to catch errors from server for eg: if the underlying resource isn't present / serving or eventing isn't installed.

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve
thanks!

**Pull request merged**